Trouble with Schema generation

boolean isFeatured(cursor.getLong(offset + 16) != 0);
                      ^
/Users/ivanbalaksha/Documents/ilitasoft/panoramaAndroidApp/app/build/generated/source/kapt/debug/com/ilitasoft/panorama/data/entity/Panorama_Schema.java:584: error: not a statement
    boolean isFeatured(cursor.getLong(offset + 16) != 0);
                                                   ^
/Users/ivanbalaksha/Documents/ilitasoft/panoramaAndroidApp/app/build/generated/source/kapt/debug/com/ilitasoft/panorama/data/entity/Panorama_Schema.java:584: error: ';' expected
    boolean isFeatured(cursor.getLong(offset + 16) != 0);
                                                       ^
/Users/ivanbalaksha/Documents/ilitasoft/panoramaAndroidApp/app/build/generated/source/kapt/debug/com/ilitasoft/panorama/data/entity/Panorama_Schema.java:585: error: ';' expected
    boolean isPopular(cursor.getLong(offset + 17) != 0);
                     ^
/Users/ivanbalaksha/Documents/ilitasoft/panoramaAndroidApp/app/build/generated/source/kapt/debug/com/ilitasoft/panorama/data/entity/Panorama_Schema.java:585: error: not a statement
    boolean isPopular(cursor.getLong(offset + 17) != 0);
                                                  ^
/Users/ivanbalaksha/Documents/ilitasoft/panoramaAndroidApp/app/build/generated/source/kapt/debug/com/ilitasoft/panorama/data/entity/Panorama_Schema.java:585: error: ';' expected
    boolean isPopular(cursor.getLong(offset + 17) != 0);
                                                      ^
/Users/ivanbalaksha/Documents/ilitasoft/panoramaAndroidApp/app/build/generated/source/kapt/debug/com/ilitasoft/panorama/data/entity/Panorama_Schema.java:586: error: ';' expected
    boolean isRecent(cursor.getLong(offset + 18) != 0);
                    ^
/Users/ivanbalaksha/Documents/ilitasoft/panoramaAndroidApp/app/build/generated/source/kapt/debug/com/ilitasoft/panorama/data/entity/Panorama_Schema.java:586: error: not a statement
    boolean isRecent(cursor.getLong(offset + 18) != 0);
                                                 ^
/Users/ivanbalaksha/Documents/ilitasoft/panoramaAndroidApp/app/build/generated/source/kapt/debug/com/ilitasoft/panorama/data/entity/Panorama_Schema.java:586: error: ';' expected
    boolean isRecent(cursor.getLong(offset + 18) != 0);
                                                     ^
/Users/ivanbalaksha/Documents/ilitasoft/panoramaAndroidApp/app/build/generated/source/kapt/debug/com/ilitasoft/panorama/data/entity/Panorama_Schema.java:587: error: ';' expected
    boolean isNearby(cursor.getLong(offset + 19) != 0);
                    ^
/Users/ivanbalaksha/Documents/ilitasoft/panoramaAndroidApp/app/build/generated/source/kapt/debug/com/ilitasoft/panorama/data/entity/Panorama_Schema.java:587: error: not a statement
    boolean isNearby(cursor.getLong(offset + 19) != 0);
                                                 ^
/Users/ivanbalaksha/Documents/ilitasoft/panoramaAndroidApp/app/build/generated/source/kapt/debug/com/ilitasoft/panorama/data/entity/Panorama_Schema.java:587: error: ';' expected
    boolean isNearby(cursor.getLong(offset + 19) != 0);
                                                     ^
12 errors

my class:

```kotlin
@Table
class Panorama {
    @PrimaryKey val id: Int
    @Column val title: String
    @Column val folder: String
    @Column val photography: String
    @Column val views: Int
    @Column val countryId: Int
    @Column val continentId: Int
    @Column val regionId: Int
    @Column val cityId: Int
    @Column val rating: Int
    @Column val commercial: Int
    @Column val favourite: Int
    @Column(indexed = true) val ltd: Float
    @Column(indexed = true) val lgt: Float
    @Column val athDelta: Float
    @Column val previewSmall: String
    @Column val previewBig: String
    @Column(indexed = true) val isFeatured: Boolean
    @Column(indexed = true) val isPopular: Boolean
    @Column(indexed = true) val isRecent: Boolean
    @Column(indexed = true) val isNearby: Boolean
    @Column val user: User?


    constructor(@Setter("id") id: Int,
                @Setter("title") title: String,
                @Setter("folder") folder: String,
                @Setter("photography") photography: String,
                @Setter("views") views: Int,
                @Setter("countryId") countryId: Int,
                @Setter("continentId") continentId: Int,
                @Setter("regionId") regionId: Int,
                @Setter("cityId") cityId: Int,
                @Setter("rating") rating: Int,
                @Setter("commercial") commercial: Int,
                @Setter("favourite") favourite: Int,
                @Setter("ltd") ltd: Float,
                @Setter("lgt") lgt: Float,
                @Setter("athDelta") athDelta: Float,
                @Setter("previewSmall") previewSmall: String,
                @Setter("previewBig") previewBig: String,
                @Setter("isFeatured") isFeatured: Boolean,
                @Setter("isPopular") isPopular: Boolean,
                @Setter("isRecent") isRecent: Boolean,
                @Setter("isNearby") isNearby: Boolean,
                @Setter("user") user: User?) {
        this.id = id
        this.title = title
        this.folder = folder
        this.photography = photography
        this.views = views
        this.countryId = countryId
        this.continentId = continentId
        this.regionId = regionId
        this.cityId = cityId
        this.rating = rating
        this.commercial = commercial
        this.favourite = favourite
        this.ltd = ltd
        this.lgt = lgt
        this.athDelta = athDelta
        this.previewSmall = previewSmall
        this.previewBig = previewBig
        this.isFeatured = isFeatured
        this.isPopular = isPopular
        this.isRecent = isRecent
        this.isNearby = isNearby
        this.user = user
    }
}
```

--- a/library/src/test/java/com/github/gfx/android/orma/test/model/ModelIssue248.java
+++ b/library/src/test/java/com/github/gfx/android/orma/test/model/ModelIssue248.java
@@ -1,0 +1,38 @@
+/*
+ * Copyright (c) 2015 FUJI Goro (gfx).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.github.gfx.android.orma.test.model;
+
+import com.github.gfx.android.orma.annotation.Column;
+import com.github.gfx.android.orma.annotation.Table;
+
+/**
+ * https://github.com/gfx/Android-Orma/issues/248
+ */
+@Table
+public class ModelIssue248 {
+
+    @Column
+    private boolean isFoo;
+
+    public void isFoo(boolean value) {
+        isFoo = value;
+    }
+
+    public boolean isFoo() {
+        return isFoo;
+    }
+}

--- a/processor/src/main/java/com/github/gfx/android/orma/processor/model/SchemaDefinition.java
+++ b/processor/src/main/java/com/github/gfx/android/orma/processor/model/SchemaDefinition.java
@@ -217,8 +217,8 @@ public class SchemaDefinition {
                     Getter getter = methodElement.getAnnotation(Getter.class);
                     Setter setter = methodElement.getAnnotation(Setter.class);
 
-                    getters.put(extractNameFromGetter(getter, methodElement), methodElement);
-                    setters.put(extractNameFromSetter(setter, methodElement), methodElement);
+                    extractNameFromGetter(getters, getter, methodElement);
+                    extractNameFromSetter(setters, setter, methodElement);
                 });
 
         columns.addAll(columnElements.stream()
@@ -231,35 +231,33 @@ public class SchemaDefinition {
         return columns;
     }
 
-    private String extractNameFromGetter(Getter annotation, ExecutableElement getterElement) {
+    private void extractNameFromGetter(Map<String, ExecutableElement> map, Getter annotation, ExecutableElement accessor) {
         if (annotation != null && !Strings.isEmpty(annotation.value())) {
-            return annotation.value();
-        } else {
-            String name = getterElement.getSimpleName().toString();
-            if (isBooleanType(getterElement.getReturnType())) {
+            map.put(annotation.value(), accessor);
+        } else if (accessor.getParameters().isEmpty()) {
+            String name = accessor.getSimpleName().toString();
+            if (isBooleanType(accessor.getReturnType())) {
                 if (name.startsWith("is")) {
-                    return name.substring("is".length());
+                    map.put(name.substring("is".length()), accessor);
                 }
-                // fallback
             }
             if (name.startsWith("get")) {
-                return name.substring("get".length());
-            } else {
-                return name;
+                map.put(name.substring("get".length()), accessor);
             }
+            map.put(name, accessor);
         }
     }
 
-    private String extractNameFromSetter(Setter annotation, ExecutableElement setterElement) {
+    private void extractNameFromSetter(Map<String, ExecutableElement> map, Setter annotation, ExecutableElement accessor) {
         if (annotation != null && !Strings.isEmpty(annotation.value())) {
-            return annotation.value();
-        } else {
-            String name = setterElement.getSimpleName().toString();
+            map.put(annotation.value(), accessor);
+        } else if (accessor.getParameters().size() == 1) {
+            String name = accessor.getSimpleName().toString();
             if (name.startsWith("set")) {
-                return name.substring("set".length());
-            } else {
-                return name;
+                map.put(name.substring("set".length()), accessor);
             }
+            map.put(name, accessor);
+
         }
     }
 

